react/ProductSearchContextProvider.js

```diff
@@ -129,11 +146,7 @@ class ProductSearchContextProvider extends Component {
           const data = searchQueryProps.data || {}
           return (
             <DataLayerApolloWrapper
-              getData={() =>
-                this.getData({
-                  ...data.search,
-                })
-              }
+              getData={() => this.getData(data.search)}
```

Comment:
You've changed the semantics of the parameter. If `data.search` doesn't return anything, the parameter will be null or undefined, but, the previous way, will return an empty object.

Reply:
If we want the parameter to always return something truthy, what [these lines](https://github.com/vtex-apps/store/blob/9dbf0e84a0b94afc23374b221d0af600b6c27016/react/ProductSearchContextProvider.js#L78-L80) are for?

Reply:
💯